util/passport.js

```diff
@@ -31,6 +36,73 @@ module.exports = function(passport) {
         });
     }));
 
+    passport.use(new GoogleStrategy({
+        clientID: config.google.clientID,
+        clientSecret: config.google.clientSecret,
+        callbackURL: config.baseURL + "/auth/google/callback"
+    }, function(req, accessToken, refreshToken, profile, done) {
+        function renderResponse(errorMsg) {
+            return responseFactory.sendRenderedResponse("public/signup", req, req.res, { captcha: app.recaptcha.render(), error: { message: errorMsg || "An unknown error occurred" }, username: "" });
+        }
+        User.findOne({ name: "google_" + profile.id }, function(err, user) {
```

Comment:
I still think that we should prompt the user to make their own username, after they sign in for the first time, and then use that rather than the service + ID.